### DISCUSSION TITLE
Tasmota Thermostat

### DISCUSSION THREAD

**Discussion by Prinsessen** [arendst/Tasmota#16634] [Q&A] [answered]
Can anyone explain how I define the internal temperature sensor to use ADC_TEMP GPI17

In the source code its default to this:
#define USE_DS18x20

**Comment by barbudor** [comment, marked as answer]
Not exactly. The line you are pointing is about having the driver for BS18x20 in the build

What you are looking for is documented in [here](https://tasmota.github.io/docs/Thermostat/#local-temperature-sensor)
The default is set by
```
#define THERMOSTAT_SENSOR_NAME                "DS18B20" // Name of the local sensor to be used
```
So you should have the following lines in your `user_config_override.h`:
```
#ifdef THERMOSTAT_SENSOR_NAME
#undef THERMOSTAT_SENSOR_NAME
#endif
#define THERMOSTAT_SENSOR_NAME                "ANALOG"
```
This will tell the thermostat to use the value from `"ANALOG":{"Temperature":00.0}` in the SENSOR message

Note that also as per the documentation you need to enter the command `SENSORINPUTSET 1` to specify using the local sensor instead of a value coming from MQTT.

**Reply by Prinsessen** [reply]
Works perfect - thanks 😊
